internal/secrets/localsecrets/localsecrets.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package localsecrets
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"github.com/google/go-cloud/internal/secrets/driver"
+	"golang.org/x/crypto/nacl/secretbox"
+	"io"
+)
+
+// secretKeeper holds a secret for use in symmetric encryption,
+// and implementations of driver.Encryper and driver.Decrypter.
+type secretKeeper struct {
+	secretKey [32]byte
+	Encrypter driver.Encrypter
+	Decrypter driver.Decrypter
+}
+
+// NewSecretKeeper takes a secret key and returns a secretKeeper.
+func NewSecretKeeper(sk string) (*secretKeeper, error) {
+	skb := []byte(sk)
+	enc := base64.StdEncoding
+	dst := make([]byte, enc.EncodedLen(len(skb)))
+	enc.Encode(dst, skb)
+
+	var dst32 [32]byte
+	copy(dst32[:], dst)
+
+	skr := &secretKeeper{secretKey: dst32}
```

Comment:
skr := &secretKeeper{
   secretKey: ...,
   Encrypter: ...,
   Decrypter: ...,
}

Reply:
It is the way it is currently because I assumed I cannot pass a pointer to the object that I'm creating in the composite literal to create it. Is this incorrect?